discovery/refresh/refresh.go

```diff
@@ -114,7 +114,9 @@ func (d *Discovery) Run(ctx context.Context, ch chan<- []*targetgroup.Group) {
 
 func (d *Discovery) refresh(ctx context.Context) ([]*targetgroup.Group, error) {
 	now := time.Now()
-	defer d.duration.Observe(time.Since(now).Seconds())
+	func() {
+		defer d.duration.Observe(time.Since(now).Seconds())
```

Comment:
Shouldn't the `defer` be outside of the `func()`?

Reply:
Thanks, fixed!